src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+              for (const choiceValue of plan) {
+                runChoice(choiceValue);
+              }
+              if (fightingSteve || currentRound()) runCombat();
+            },
+            true
+          );
+        }
+        return fightingSteve;
+      }
+      return false;
+    },
+  },
+  {
+    name: "Proton Ghost",
+    available: () =>
+      sober() &&
+      have($item`protonic accelerator pack`) &&
+      get("questPAGhost") !== "unstarted" &&
+      !!get("ghostLocation"),
+    perform: () => {
+      const ghostLocation = get("ghostLocation");
+      if (!ghostLocation) return false;
+      freeFightPrep(
+        new Requirement(
+          ghostLocation === $location`The Icy Peak` ? ["Cold Resistance 5 min"] : [],
+          {
+            forceEquip: $items`protonic accelerator pack`,
+          }
+        )
+      );
+      adventureMacro(ghostLocation, Macro.ghostBustin());
+      return get("questPAGhost") === "unstarted";
+    },
+  },
+  {
+    name: "Vote Wanderer",
+    available: () =>
+      sober() &&
+      have($item`"I Voted!" sticker`) &&
+      totalTurnsPlayed() % 11 === 1 &&
+      get("lastVoteMonsterTurn") < totalTurnsPlayed() &&
+      get("_voteFreeFights") < 3,
+    perform: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`"I Voted!" sticker` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return get("lastVoteMonsterTurn") === totalTurnsPlayed();
+    },
+  },
+  {
+    name: "Digitize Wanderer",
+    available: () => Counter.get("Digitize Monster") <= 0,
+    perform: () => {
+      const isEmbezzler = get("_sourceTerminalDigitizeMonster") === embezzler;
+      const start = get("_sourceTerminalDigitizeMonsterCount");
+
+      const shouldGoUnderwater =
+        isEmbezzler &&
+        !get("_envyfishEggUsed") &&
+        myLevel() >= 11 &&
+        (getModifier("Adventure Underwater") ||
+          waterBreathingEquipment.some((item) => have(item))) &&
+        (getModifier("Underwater Familiar") ||
+          familiarWaterBreathingEquipment.some((item) => have(item))) &&
+        (have($effect`Fishy`) || (have($item`fishy pipe`) && !get("_fishyPipeUsed"))) &&
+        !have($item`envyfish egg`) &&
+        canAdventure($location`The Briny Deeps`);
+      const targetLocation = shouldGoUnderwater
+        ? $location`The Briny Deeps`
+        : determineDraggableZoneAndEnsureAccess();
+
+      isEmbezzler ? embezzlerPrep() : freeFightPrep();
+      adventureMacroAuto(targetLocation, Macro.basicCombat());
+      return get("_sourceTerminalDigitizeMonsterCount") !== start;
+    },
+  },
+  {
+    name: "Guaranteed Kramco",
+    available: () => sober() && kramcoGuaranteed(),
+    perform: () => {
+      freeFightPrep(new Requirement([], { forceEquip: $items`Kramco Sausage-o-Matic™` }));
+      adventureMacroAuto(determineDraggableZoneAndEnsureAccess(), Macro.basicCombat());
+      return !kramcoGuaranteed();
```

Comment:
what is this return actually doing for us in any way

Reply:
This might not actually be useful; I want to have a check to make sure people don't end up in some sort of weird infinite loop, where they try to take an action but never actually do so, thereby never changing the value of the `available` function. So the idea is we return a value that tells us "did we end up going through with this?"

A failure here could range from "jesus christ what" to "oh, yeah, hiccup, just do the next thing." For the most part, we just do the next thing here.

Reply:
we should put a warning out when a task fails i think

Reply:
Just printing, or erroring out? I don't think we should error out just because there are enough dumb cases where we hit zone intro adventures.